components/elm/src/cpl/lnd_import_export.F90

```diff
@@ -751,7 +751,7 @@ subroutine lnd_import( bounds, x2l, atm2lnd_vars, glc2lnd_vars, lnd2atm_vars)
               call mpi_bcast (smap05_lat, 360, MPI_REAL8, 0, mpicom, ier)
             end if
           end if
-        end if
+        !end if
```

Comment:
If this "end if" can be removed then how did it compile without error before?

Reply:
This extra "end if" does cause a compilation error on Chrysalis when setting up the case by script (see below). However, if typing "./case.build" in the case folder, the compilation can pass. @evasinha  and @dmricciuto also experienced the same issue.

/gpfs/fs1/home/ac.ztan/E3SM_erosion/components/elm/src/cpl/lnd_import_export.F90(754): error #6606: The block construct names must match, and they do not.
    end if
^
/gpfs/fs1/home/ac.ztan/E3SM_erosion/components/elm/src/cpl/lnd_import_export.F90(928): error #6317: An ENDIF occurred without a corresponding IF THEN or ELSE statement.
    end if model_filter
-------^
compilation aborted for /gpfs/fs1/home/ac.ztan/E3SM_erosion/components/elm/src/cpl/lnd_import_export.F90 (code 1)

Reply:
But in the land unit test, both baseline code with this line and my code without this line can be compiled without errors.

Reply:
@rljacob This part of the code is within a `#ifdef CPL_BYPASS` at [lnd_import_export.F90#L199
](https://github.com/E3SM-Project/E3SM/blob/1c8a5eb5b6b80be9e186da3e2514d51056450dac/components/elm/src/cpl/lnd_import_export.F90#L199) and we don't have a test for it, so we never caught it.

@tanzeli1982 Remove the `!endif` and shift the indentation of the code block [lnd_import_export.F90#L711-L753](https://github.com/E3SM-Project/E3SM/blob/1c8a5eb5b6b80be9e186da3e2514d51056450dac/components/elm/src/cpl/lnd_import_export.F90#L711-L753) to the left.

Reply:
@tanzeli1982 I have made some changes related to the topounit subgrid structure.